### PR TITLE
INFRA-332-userregistry-aks-init-cluster

### PR DESCRIPTION
**Pull request by diegolagospagopa**
### List of changes

aks first setup

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [x] Remove existing resources

### Env to apply

- [x] DEV
- [ ] UAT
- [ ] PROD

### CODE REVIEW

File: src/.env/prod/terraform.tfvars [hunk 1]
```diff
@@ -1,40 +1,78 @@
...
 location = "westeurope"
-
 tags = {
   CreatedBy   = "Terraform"
   Environment = "Prod"
   Owner       = "usrreg"
   Source      = "https://github.com/pagopa/usrreg-infra"
   CostCenter  = "TS310 - PAGAMENTI & SERVIZI"
 }
+lock_enable = false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
lock_enable = true
```

File: src/.env/prod/terraform.tfvars [hunk 2]
```diff
@@ -1,40 +1,78 @@
...
+cidr_vnet               = ["10.1.0.0/16"]
+cidr_subnet_azdoa       = ["10.1.130.0/24"]
+cidr_subnet_postgres    = ["10.1.129.0/24"]
+cidr_subnet_appgateway  = ["10.1.128.0/24"]
+cidr_subnet_apim        = ["10.1.136.0/24"]
 
 # dns
 external_domain = "pagopa.it"
-dns_zone_prefix = "userregistry"
+dns_zone_prefix = "prod.userregistry"
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars:26]:
```suggestion
dns_zone_prefix = "userregistry"
```

File: src/.env/prod/terraform.tfvars [hunk 3]
```diff
@@ -1,40 +1,78 @@
...
 
 # azure devops
 azdo_sp_tls_cert_enabled = true
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
```

File: src/.env/dev/terraform.tfvars [hunk 4]
```diff
@@ -1,41 +1,79 @@
...
+cidr_subnet_apim        = ["10.1.136.0/24"]
+cidr_subnet_k8s         = ["10.1.0.0/17"]
 
 # dns
-external_domain = "pagopa.it"
 dns_zone_prefix = "dev.userregistry"
+external_domain = "pagopa.it"
 
 # azure devops
 azdo_sp_tls_cert_enabled = true
```
Comment (pasqualedevita) [inline src/.env/dev/terraform.tfvars]:
```suggestion
```

File: src/.env/prod/terraform.tfvars [hunk 5]
```diff
@@ -1,40 +1,78 @@
...
 enable_azdoa             = true
 enable_iac_pipeline      = true
 
-# apim
-
+# ❇️ app_gateway
+app_gateway_sku_name = "Standard_v2"
+app_gateway_sku_tier = "Standard_v2"
+app_gateway_api_certificate_name = "api-prod-userregistry-pagopa-it"
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
app_gateway_api_certificate_name = "api-userregistry-pagopa-it"
```

File: src/.env/prod/terraform.tfvars [hunk 6]
```diff
@@ -1,40 +1,78 @@
...
 # azure devops
 azdo_sp_tls_cert_enabled = true
 enable_azdoa             = true
 enable_iac_pipeline      = true
 
-# apim
-
+# ❇️ app_gateway
+app_gateway_sku_name = "Standard_v2"
+app_gateway_sku_tier = "Standard_v2"
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
app_gateway_sku_name = "Standard_v2" #  todo change to WAF_v2
app_gateway_sku_tier = "Standard_v2" #  todo change to WAF_v2
```

File: src/.env/prod/terraform.tfvars [hunk 7]
```diff
@@ -1,40 +1,78 @@
...
+app_gateway_api_certificate_name = "api-prod-userregistry-pagopa-it"
 
-# app_gateway
+app_gateway_alerts_enabled=false
+app_gateway_waf_enabled=false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
app_gateway_alerts_enabled=true
app_gateway_waf_enabled=false # todo change to true
```

File: src/.env/prod/terraform.tfvars [hunk 8]
```diff
@@ -1,40 +1,78 @@
...
 
+# 🗄 postgresql
+postgres_sku_name       = "GP_Gen5_2"
+postgres_enable_replica = false
+postgres_public_network_access_enabled = true
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
postgres_public_network_access_enabled = false
```

File: src/.env/prod/terraform.tfvars [hunk 9]
```diff
@@ -1,40 +1,78 @@
...
+app_gateway_sku_tier = "Standard_v2"
+app_gateway_api_certificate_name = "api-prod-userregistry-pagopa-it"
 
-# app_gateway
+app_gateway_alerts_enabled=false
+app_gateway_waf_enabled=false
 
+# 🗄 postgresql
+postgres_sku_name       = "GP_Gen5_2"
+postgres_enable_replica = false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars:42]:
```suggestion
```

File: src/.env/prod/terraform.tfvars [hunk 10]
```diff
@@ -1,40 +1,78 @@
...
+  log_connections             = "off"
+  log_line_prefix             = "%t [%p apps:%a host:%r]: [%l-1] db=%d,user=%u"
+  log_temp_files              = "4096"
+  maintenance_work_mem        = "524288"
+  max_wal_size                = "4096"
+  log_connections             = "on"
+  log_checkpoints             = "on"
+  connection_throttling       = "on"
+}
+postgres_alerts_enabled = false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
postgres_alerts_enabled = true
```

File: src/.env/prod/terraform.tfvars [hunk 11]
```diff
@@ -1,40 +1,78 @@
...
+  log_autovacuum_min_duration = "5000"
+  log_connections             = "off"
+  log_line_prefix             = "%t [%p apps:%a host:%r]: [%l-1] db=%d,user=%u"
+  log_temp_files              = "4096"
+  maintenance_work_mem        = "524288"
+  max_wal_size                = "4096"
+  log_connections             = "on"
+  log_checkpoints             = "on"
+  connection_throttling       = "on"
+}
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
```

File: src/.env/prod/terraform.tfvars [hunk 12]
```diff
@@ -1,40 +1,78 @@
...
+postgres_alerts_enabled = false
 
-# postgresql
-
+#
+# 🗺 APIM
+#
+apim_publisher_name = "pagoPA SelfCare DEV"
+apim_sku            = "Developer_1"
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
apim_sku            = "Developer_1" # todo change to Premium_1
```

File: src/.env/prod/terraform.tfvars [hunk 13]
```diff
@@ -1,40 +1,78 @@
...
+  connection_throttling       = "on"
+}
+postgres_alerts_enabled = false
 
-# postgresql
-
+#
+# 🗺 APIM
+#
+apim_publisher_name = "pagoPA SelfCare DEV"
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
apim_publisher_name = "pagoPA SelfCare PROD"
```

File: src/.env/prod/terraform.tfvars [hunk 14]
```diff
@@ -1,40 +1,78 @@
...
+apim_sku            = "Developer_1"
+apim_api_internal_certificate_name = "api-internal-userregistry-pagopa-it"
 
-# apps
+#
+# AKS
+#
+aks_private_cluster_enabled = false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
aks_private_cluster_enabled = true
```

File: src/.env/prod/terraform.tfvars [hunk 15]
```diff
@@ -1,40 +1,78 @@
...
+aks_alerts_enabled          = false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
aks_alerts_enabled          = true
```

File: src/.env/prod/terraform.tfvars [hunk 16]
```diff
@@ -1,40 +1,78 @@
...
+# This is the k8s ingress controller ip. It must be in the aks subnet range.
+reverse_proxy_ip            = "10.1.0.250"
+aks_max_pods                = 100
+aks_enable_auto_scaling     = false
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
aks_enable_auto_scaling     = true
```

File: src/.env/prod/terraform.tfvars [hunk 17]
```diff
@@ -1,40 +1,78 @@
...
+aks_node_min_count          = null
+aks_node_max_count          = null
```
Comment (pasqualedevita) [inline src/.env/prod/terraform.tfvars]:
```suggestion
aks_node_min_count          = 1
aks_node_max_count          = 3
```

File: src/.env/uat/terraform.tfvars [hunk 18]
```diff
@@ -1,23 +1,25 @@
 # general
 env_short = "u"
 location = "germanywestcentral"
-
 tags = {
   CreatedBy   = "Terraform"
-  Environment = "Uat"
+  Environment = "UAT"
   Owner       = "usrreg"
   Source      = "https://github.com/pagopa/usrreg-infra"
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
  Source      = "https://github.com/pagopa/userregistry-infra"
```

File: src/.env/uat/terraform.tfvars [hunk 19]
```diff
@@ -1,23 +1,25 @@
 # general
 env_short = "u"
 location = "germanywestcentral"
-
 tags = {
   CreatedBy   = "Terraform"
-  Environment = "Uat"
+  Environment = "UAT"
   Owner       = "usrreg"
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
  Owner       = "UserRegistry"
```

File: src/.env/uat/terraform.tfvars [hunk 20]
```diff
@@ -1,23 +1,25 @@
 # general
 env_short = "u"
 location = "germanywestcentral"
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
location = "northeurope"
```

File: src/.env/uat/terraform.tfvars [hunk 21]
```diff
@@ -28,13 +30,49 @@ azdo_sp_tls_cert_enabled = true
...
+app_gateway_sku_tier = "Standard_v2"
+app_gateway_api_certificate_name = "api-uat-userregistry-pagopa-it"
 
-# app_gateway
+app_gateway_alerts_enabled=false
+app_gateway_waf_enabled=false
 
+# 🗄 postgresql
+postgres_sku_name       = "GP_Gen5_2"
+postgres_enable_replica = false
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars:43]:
```suggestion
```

File: src/.env/uat/terraform.tfvars [hunk 22]
```diff
@@ -28,13 +30,49 @@ azdo_sp_tls_cert_enabled = true
...
+  log_autovacuum_min_duration = "5000"
+  log_connections             = "off"
+  log_line_prefix             = "%t [%p apps:%a host:%r]: [%l-1] db=%d,user=%u"
+  log_temp_files              = "4096"
+  maintenance_work_mem        = "524288"
+  max_wal_size                = "4096"
+  log_connections             = "on"
+  log_checkpoints             = "on"
+  connection_throttling       = "on"
+}
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
```

File: src/.env/uat/terraform.tfvars [hunk 23]
```diff
@@ -28,13 +30,49 @@ azdo_sp_tls_cert_enabled = true
...
+postgres_alerts_enabled = false
 
-# postgresql
-
+#
+# 🗺 APIM
+#
+apim_publisher_name = "pagoPA SelfCare DEV"
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
apim_publisher_name = "pagoPA SelfCare UAT"
```

File: src/.env/uat/terraform.tfvars [hunk 24]
```diff
@@ -28,13 +30,49 @@ azdo_sp_tls_cert_enabled = true
...
-# apps
+#
+# AKS
+#
+aks_private_cluster_enabled = false
+aks_alerts_enabled          = false
+# This is the k8s ingress controller ip. It must be in the aks subnet range.
+reverse_proxy_ip            = "10.1.0.250"
+aks_max_pods                = 100
+aks_enable_auto_scaling     = false
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
aks_enable_auto_scaling     = true
```

File: src/.env/uat/terraform.tfvars [hunk 25]
```diff
@@ -28,13 +30,49 @@ azdo_sp_tls_cert_enabled = true
...
+aks_node_min_count          = null
+aks_node_max_count          = null
```
Comment (pasqualedevita) [inline src/.env/uat/terraform.tfvars]:
```suggestion
aks_node_min_count          = 1
aks_node_max_count          = 1
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/.env/prod/terraform.tfvars (hunk 6 above)
Comment (diegolagospagopa) [reply src/.env/prod/terraform.tfvars]: Done

File: src/.env/prod/terraform.tfvars (hunk 3 above)
Comment (diegolagospagopa) [reply src/.env/prod/terraform.tfvars]: Done

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**